src/qtt/algorithms/generic.py

```diff
@@ -93,13 +97,16 @@ def subpixelmax(A, mpos, verbose=0):
     cy = val
     ay = (valm + valp) / 2 - cy
     by = ay + cy - valm
-    shift = -by / (2 * ay)  # Maxima of quadradic
+
+    if np.any(ay == 0):
+        shift = 0 * ay
+    else:
+        shift = -by / (2 * ay)  # Maxima of quadradic
 
     if verbose:
         print('subpixelmax: mp %d, pp %d\n', mp, pp)
         print('subpixelmax: ap %.3f, by %.3f , cy %.3f\n', ay, by, cy)
```

Comment:
```suggestion
        print('subpixelmax: ap %.3f, by %.3f , cy %.3f\n' % (ay, by, cy))
```

Reply:
There is another print. I cannot make a suggestion for it but it should be: 

print('subpixelmax1d: shift %.3f\n' % shift)